### PR TITLE
Added wrapper function handling for geometry_in and geography_in functions

### PR DESCRIPTION
**Pull request by Anikait143**
### 1. Issues
- TDS - Receiver function for GEOGRAPHY datatype does not work as expected if the -90 < Longitude < 90 in JDBC framework
- Incorrect conversion of procedure arguments when the input argument is of Geography datatype

### 2. Changes made to fix the issues
- For longitude issue, we saw that the PostGIS functions was itself bounding the value of the longitude, so, we added a wrapper function to handle this.
- Fixed Cast functions for char, varchar, bytea to geometry and geography to correctly store point instance and avoid truncating of longitude values
```
# Input
prepst#!#exec#!#GEOGRAPHY|-|location|-|Point(290.345 45.1234):4326
```
```
# previous output (wrong output)
select location.Long from SPATIALPOINTGEOG_dt
-69.65499999999997
```
```
# current output (correct output)
select location.Long from SPATIALPOINTGEOG_dt
290.345
```

- Added wrapper function handling for geometry_in and geography_in functions to support direct invoke through Point text
```
# previous output (wrong output)
1> create procedure p1 @a geography as select @a.Lat as Latitude, @a.Long as Longitude
2> go
1> p1 'POINT(1 2)'
2> go
Latitude                 Longitude               
------------------------ ------------------------
                     1.0                      2.0(1 rows affected)

1> p1 'POINT(1 200)'
2> go
Latitude                 Longitude               
------------------------ ------------------------
                     1.0                    -20.0(1 rows affected)

1> p1 'POINT(1000 20)'
2> go
Latitude                 Longitude               
------------------------ ------------------------
                   -80.0                     20.0(1 rows affected)
1> 
```
```
# current output (correct output)
1> create procedure p1 @a geography as select @a.Lat as Latitude, @a.Long as Longitude
2> go
1> p1 'POINT(1 2)'
2> go
Latitude                 Longitude               
------------------------ ------------------------
                     2.0                      1.0(1 rows affected)

1> p1 'POINT(1 200)'
2> go
Error converting data type varchar to geography.

1> p1 'POINT(1000 20)'
2> go
Latitude                 Longitude               
------------------------ ------------------------
                    20.0                   1000.0(1 rows affected)
1> 
```
- Added Test Cases for these issues
- Corrected Order by clause in text to resolve pipeline failures
- Added logical replication tests for Geospatial datatypes

Task: BABEL- 4753, BABEL-4757
Authored-by: Anikait Agrawal [agraani@amazon.com](mailto:agraani@amazon.com)
Signed-off-by: Anikait Agrawal [agraani@amazon.com](mailto:agraani@amazon.com)

### Test Scenarios Covered ###
* **Use case based -**  TestSpatialPoint-vu-*, datatype_test

* **Boundary conditions -**

* **Arbitrary inputs -**

* **Negative test cases -**

* **Minor version upgrade tests -** 

* **Major version upgrade tests -** 

* **Performance tests -**

* **Tooling impact -**

* **Client tests -**

### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the Apache 2.0 and PostgreSQL licenses, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: contrib/babelfishpg_common/sql/geometry.sql [hunk 1]
```diff
@@ -1,6 +1,6 @@
 CREATE OR REPLACE FUNCTION sys.geometryin(cstring)
     RETURNS sys.GEOMETRY
-    AS '$libdir/postgis-3', 'LWGEOM_in'
+    AS 'babelfishpg_common', 'LWGEOM_in'
```
Comment (rishabhtanwar29) [inline contrib/babelfishpg_common/sql/geometry.sql]: Let's re-name the C function as `geometry_in`

File: test/JDBC/expected/datatype_test.out [hunk 2]
```diff
@@ -122,8 +126,8 @@ GO
...
+bigint#!#binary#!#bit#!#char#!#date#!#datetime#!#datetime2#!#datetimeoffset#!#numeric#!#float#!#image#!#int#!#money#!#nchar#!#ntext#!#numeric#!#nvarchar#!#real#!#smalldatetime#!#smallint#!#smallmoney#!#sql_variant#!#varchar#!#text#!#time#!#tinyint#!#uniqueidentifier#!#varbinary#!#varchar#!#xml#!#geometry#!#geography
+1#!#1A00000000000000000000000000000000000000#!#1#!#abc                 #!#2000-06-01#!#2000-06-01 00:00:00.0#!#2000-06-01 00:00:00.0000000#!#2000-06-01 00:00:00.0000000 +00:00#!#8.4230#!#1.43E19#!#0456#!#1#!#50.0000#!#😊😋😎😍😅😆        #!#😊😋😎😍😅😆#!#8.4230#!#😊😋😎😍😅😆#!#1.43E19#!#2000-06-01 00:00:00.0#!#1#!#50.0000#!#123#!#abc#!#abc#!#01:01:12.0000000#!#1#!#1E984725-C51C-4BF4-9960-E1C80E27ABA0#!#01AB#!#abc#!#<xml></xml>#!#E6100000010C00000000000000000000000000000000#!#E6100000010C00000000000000000000000000000000
```
Comment (rishabhtanwar29) [inline test/JDBC/expected/datatype_test.out]: Let's print geography and geometry columns as text (STAsText)

File: contrib/babelfishpg_common/sql/geometry.sql (hunk 1 above)
Comment (Anikait143) [reply contrib/babelfishpg_common/sql/geometry.sql]: Done

File: test/JDBC/expected/datatype_test.out (hunk 2 above)
Comment (Anikait143) [reply test/JDBC/expected/datatype_test.out]: Done

**Review by jsudrik — APPROVED** [review]
Identical changes to 3_x. Approved.

**Pull request merged**